recipes/cppkafka/all/conanfile.py

```diff
@@ -0,0 +1,137 @@
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.33.0"
+
+class CppKafkaConan(ConanFile):
+    name = "cppkafka"
+    description = "Modern C++ Apache Kafka client library (wrapper for librdkafka)"
+    topics = ("librdkafka", "kafka")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/mfontanini/cppkafka"
+    license = "MIT"
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "cmake", "cmake_find_package_multi"
+
+    options = {
+       "shared": [True, False],
+       "fPIC": [True, False],
+       "multithreaded": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "multithreaded": True,
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        self.options["librdkafka"].shared = self.options.shared
+
+    def requirements(self):
+        self.requires("boost/1.78.0")
+        self.requires("librdkafka/1.8.2")
+
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "gcc": "6",
+            "clang": "4",
+            "apple-clang": "5.1",
+        }
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, 14)
+
+        minimum_version = self._minimum_compilers_version.get(str(self.settings.compiler), False)
+        if not minimum_version:
+            self.output.warn(
+                "{}/{} requires C++14. Your compiler is unknown. Assuming it supports C++14.".format(self.name, self.version,))
+        elif tools.Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                "{}/{} requires C++14, which your compiler does not support.".format(self.name, self.version,))
+
+        if self.settings.compiler in ["clang", "apple-clang"] and self.settings.compiler.get_safe("libcxx") == "libc++":
+            raise ConanInvalidConfiguration(
+                "{}/{} doesn't support {} with libc++".format(self.name, self.version, self.settings.compiler))
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        opts = dict()
+        opts["RDKAFKA_LIBRARY"] = self.deps_cpp_info["librdkafka"].rootpath
```

Comment:
why don't you use only cmake.definitions? it will be the same thing at the end

Reply:
You are right.
I forget to delete these lines.